lib/kinjen.groovy

```diff
@@ -173,28 +173,59 @@ static pg_package_stage( script )
 
 /**
  * The basic database import task.
+ * Does nothing if there has been no change in the database directory since the last successful build.
+ * We only go back 25 commits when looking for a successful build.
+ * If a successful build can not be found for the past 25 commits (or back until a commit was part of master)
+ * then a comparison is done against master to determine if this stage should run.
  */
 static import_stage( script )
 {
-  /*
-    Future: Only run import if there are db changes since the last time import successfully ran
-    Now: Only run import if there are db changes on this branch (compared to master)
-  */
-
-  // Future: find last commit with successful import
-  // Now: just fall back on the master branch
-  def compare = 'origin/master'
-
-  // find changes in db directory since target commit
-  def changes = script.sh(script: "git diff --name-only ${compare} database", returnStdout: true).trim()
+  // find changes in db directory compared to master
+  def changes = script.sh(script: "git diff --name-only origin/master database", returnStdout: true).trim()
   if ( changes ) {
-    script.echo 'Db import stage is necessary due to changes in the database directory'
+    script.echo 'There are changes in the database directory, compared to master'
     script.echo changes
-    script.stage( 'DB Import' ) {
-      script.sh 'xvfb-run -a bundle exec buildr ci:import'
+
+    // Is there a successful build we can use instead
+    def build_required = true
+    def stop_looking = false
+    def previous_commits = script.sh(script: "git rev-list HEAD~25..HEAD~1", returnStdout: true).trim().split()
```

Comment:
Unsure how many commits we pull into the build. At one stage we used to only have a handful of commits (1?) that we fetched during build and thus this setting may be effectively ignored...